### ISSUE TITLE
[ADAP-568] [Bug] sql_header not working with copy_partitions=True

### ISSUE THREAD

**Issue by itestyoy** [dbt-labs/dbt-bigquery#733]
### Is this a new bug in dbt-bigquery?

- [X] I believe this is a new bug in dbt-bigquery
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

```
-- models/model_one.sql

{{ 
    config(
        incremental_strategy = 'insert_overwrite',
        partition_by={
            "field": "event_date",
            "data_type": "datetime",
            "granularity": "day",
            "time_ingestion_partitioning": true,
            "copy_partitions": true
        }
    )
}}

{% call set_sql_header(config) %}
    CREATE TEMPORARY FUNCTION 
                int_to_string(version int)
        RETURNS STRING AS (
              cast(version as string)
        );
{%- endcall %}

select cast('2022-01-01' as datetime) as event_date,  int_to_string(1) as col1
```

```
18:51:31  Completed with 1 error and 0 warnings:
18:51:31  
18:51:31  Database Error in model model_one (models/model_one.sql)
18:51:31    Function not found: int_to_string at [7:55]

```

### Expected Behavior

- command working without errors  ```dbt run -s model_one -f ```

### Steps To Reproduce

- ```dbt run -s model_one -f ```

### Environment

```markdown
- OS: MacOS
- Python: 3.9.6
- dbt-core: 1.5.0
- dbt-bigquery: 1.5.0
```

**Comment by Fleid** [comment]
Hi @itestyoy ,

It looks like the issue is not limited to `copy_partitions=true`, but also applies to time ingestion partitioning. Looks like the UDF is not properly imported in that [statement](https://github.com/dbt-labs/dbt-bigquery/blob/7bc2af0cef1e42b625cee6d996b48cb91dc2e49c/dbt/include/bigquery/macros/materializations/incremental_strategy/time_ingestion_tables.sql#L45). @Kayrnt would you know more about that? It looks like you already [solved](https://github.com/dbt-labs/dbt-bigquery/pull/683) something recently in the same area.

**Comment by github-christophe-oudar** [comment]
Indeed it's that fix!
It's working fine on my custom build that include this fix.
Do you know why it wasn't included in 1.5.1?

This is my working docker environment:
```
ENV DBT_CORE_HASH d2f963e20ea945355db57ff3d34a318429a8bf1e
ENV DBT_BIGQUERY_HASH 7c216445f8009baa9cec4d61dd56693be1dd79fa
...
RUN pip install "git+https://github.com/dbt-labs/dbt-core.git@$DBT_CORE_HASH#egg=dbt-core&subdirectory=core" \
    "git+https://github.com/dbt-labs/dbt-bigquery.git@$DBT_BIGQUERY_HASH"
```

**Comment by Fleid** [comment]
That wasn't flagged for a back port. Not sure we have the bandwidth right now to do that.
 It should be in [1.6.0b2](https://pypi.org/project/dbt-bigquery/1.6.0b2/) and following.
